### PR TITLE
Slightly rigs a specific d20

### PR DESCRIPTION
**Pull request by Rukofamicom**
## About The Pull Request

* Adds a new subtype of the single-use die of fate which is not capable of rolling a 20 while it is still magical. Managing to hit a 20 causes the die to roll to a new side with the new side's effect, but also refreshes the magic to enable another roll.
* Replaces the reward for greed's ruin with this new die of fate, removing the final avenue of acquiring antag status through lavaland. 

## Why It's Good For The Game

Over time most of the forms of willingly becoming an antagonist have been slowly removed from lavaland but one remained. One that at first was overlooked, and then just wasn't dealt with because it wasn't nearly as often abused as all those that fell before it. After all, it was only a 5% chance, and came with a 10% chance of instantly killing you and an 85% chance of something in between instead.

**And then someone realized you can recharge single-use dies of fate and that information propagated.** So yeah, this removes the final alluring possibility of giving yourself antag via lavaland because someone always has to ruin the fun by abusing the system. All of the other fun possibilities remain with the die, as well as the ability to recharge it. 

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

https://user-images.githubusercontent.com/9547572/224019439-3a7608a5-b584-4353-962f-83e790f6c425.mp4

</details>

## Changelog
:cl:
tweak: The reward from greed's ruin can no longer grant players the ability to be wizards. All other rewards remain untouched.
/:cl:

### CODE REVIEW

**Comment by llol111** [comment]
but the funny wizard :(

**Comment by Tyranicranger4** [comment]
> what if alternatively each recharge lowered the amount of sides? it'd start at 20 and you get that single 5% chance

An item that (in it's current state and in this proposed alternative) encourages and facilitates crew roles to effectively antag roll has no place in the current iteration of Bee in my opinion, regardless of it's rarity or the potential risks involved in using it. Nothing good comes from leaving such a massive disconnect between code and policy. Let this part of the ruin die just like we did with the wishgranter.

**Comment by DatFrostys** [comment]
Just pointing out how unlikely it is to obtain all of this at the same time :
The greed ruin doesn't spawn everytime
Even then, the anomalous crystal has only 1/7 chance of being the recharge one.

Why remove the 20th side when the real problem is the anomalous crystal being able to reload it? We could just make it so it cannot do it at all. Because even if there is no 20th side, you can still :

- Have a spell book
- A slave that teleports at you
- All Access
- Permanent protection
- A strong weapon

All of this may result in heavy powergaming and/or self antag. By itself it isn't that bad due to the rarity, but the fact you may be able to mass produce the effects thanks to reload IS the problem.

TLDR : Blacklist die of fate from the list of reloadable items instead of removing the 20th side

**Comment by Rukofamicom** [comment]
image

I have had a slight change of method. You can now roll a 20 again instead of completely preventing it. 
*But the die rolls over to a new side*

In the event this occurs it also enables the die to be rolled for an effect once again, so there is *some* benefit to rolling a 20. You get to use the otherwise single-use die twice.

**Comment by tontyGH** [comment]
i have a suggestion

https://user-images.githubusercontent.com/39193182/224183845-14842b79-44d7-426f-8712-8ceb1fada2c2.mp4

**Comment by Howluinb** [comment]
you could just remove the ability to recharge it and leave the one gimmick item that is genuinely funny.
you could maybe even up the chances of dying, instead of only 1/10
but removing the chance to become wizard, which is, in fact, a rare event, is not bad.
i'm not gonna lie, in 800 hours on bee i've literally never seen a die of fate rolled wiz unless abuse was in play. It's way too rare for how uncommon the ruin even is in the first place

**Comment by Rukofamicom** [comment]
> you could just remove the ability to recharge it

I could yes, but that is a much more fun and meaningful interaction to remove. 

I much prefer the ability to refresh the die for its other multitude of effects than keeping it truly single use with one of the options still including wizard

**Comment by tontyGH** [comment]
https://user-images.githubusercontent.com/39193182/224331524-2df05b8c-e42c-41df-abc0-be89bb9c8500.MOV

**Comment by Hibabo** [comment]
Ruko coming in hot with another balance change that ruins fun while hardly playing the game himself/herself (whichever one you are) very cool Ruko, Thank you.

god forbid people get wizard once a blue moon from being in the most boring part of the game with 0 crew interaction and practically slave work known as lavaland mining.

**Comment by Rukofamicom** [comment]
> can pretty much guarantee that the people that are heavily against this turning you into a wizard are players that never touch lavaland in general.

Shaft miner remains my most played job on the server @Hibado. *silicon is quickly catching up though*

image

I make the changes I do to miner **because** it is my most played job and I don't think it fits well into the game dynamic:
 * A job that is almost completely solo and alone is not a good fit for a multiplayer game, so I push for changes that involve them more with the station, both in the form of pushing to bring lavaland "closer" to the station through public or semi-public access to it, and also through pushing for miners to rely on medbay when they get injured
 *  A job that draws people in with the sheer power curve compared to the station is not good for the game. Miner antagonists are a menace that isn't fun for anyone but the miner, and this is compounded by the fact that for a very long time the average miner didn't give two shits about RP or others' enjoyment of the game and were just seeking to engage in a one-sided power fantasy. I strongly supported the change that halved the power of all their equipment so that miner gear is more comparable to security gear, and it did so without actually changing the impact of fighting fauna. 
 * A job that offers the ability to transform yourself into an antagonist attracts people who only want to be antagonists - This is why all of the other antag-creations were removed or changed. This is also compounded by the fact shaft miner is the go-to for people who only want to antag roll as it's an easy job to fake a suicide as a legitimate death in the first ten minutes. Speaking from experience as the miner who keeps up with the others and brings them back for care, it's always great when the dude who you just watched die and who knew you were there goes DNR immediately.

> another off topic thing but perhaps you should nerf mechs into the floor where they belong as they are probably the most broken thing that also requires the least effort to make in the entire code base right now, nothings more fun than having a target thats inside of a durand which makes them immune to every stun and chemical while also being a giant bullet sponge that can shoot back.

This just sounds like a you problem - of course mechs make someone hard to kill, they're intended to be defensively strong, and in exchange they make the person piloting them extremely slow and also a massive target that's easy to hit and weak to EMP. Additionally if your target is just parading around a decked out combat mech as a non-antag for no reason, you should probably bring that up to admins. 

That aside, if you care more about making your target dead than interacting with them and primarily rely on reagents and stuns to do so, you sure do sound like the fun kind of antagonist player everyone enjoys having around.

**Comment by Hibabo** [comment]
> Shaft miner remains my most played job on the server @hibado. _silicon is quickly catching up though_

Do you just not want to share the actual time played or whats up with that? also i cant tell if you're trying to be funny with the @ or if you just can't read my name.

> I make the changes I do to miner **because** it is my most played job and I don't think it fits well into the game dynamic:

> * A job that is almost completely solo and alone is not a good fit for a multiplayer game, so I push for changes that involve them more with the station. I've been pushing to bring lavaland "closer" to the station through public or semi-public access to it, and also through pushing for miners to rely on medbay when they get injured.

I agree that mining has massive issues in basically isolating the people playing it from the station.

> * A job that draws people in with the sheer power curve compared to the station is not good for the game. Miner antagonists were a menace that wasn't fun for anyone but the miner, and this was compounded by the fact that for a very long time the average miner didn't give two shits about RP or others' enjoyment of the game and were just seeking to engage in a one-sided power fantasy. I strongly supported the change that halved the power of all their equipment so that miner gear is more comparable to security gear, and it did so without actually changing the impact of fighting fauna.

One of the reasons people put up with the absolute horror that is singleplayer experience ss13 is due to the unique interactions that could come from some of the gear which you basically removed from existance, The voodoo doll for example was a great tool that could also be used for RP gimmicks which you decided that miners should no longer have

Some of the stuff in there was pretty bad for RP though i agree.


> * A job that offers the ability to transform yourself into an antagonist attracts people who only want to be antagonists - This is why all of the other antag-creations were removed or changed. This is also compounded by the fact shaft miner is the go-to for people who only want to antag roll as it's an easy job to fake a suicide as a legitimate death in the first ten minutes. Speaking from experience as the miner who keeps up with the others and brings them back for care, it's always great when the dude who you just watched die and who knew you were there goes DNR immediately.

Its almost like i suggested several other changes instead of just removing it I dont have a problem with it being changed for the better but outright removing it is like I said, just removing fun.


> This just sounds like a you problem - of course mechs make someone hard to kill, they're intended to be defensively strong, and in exchange they make the person piloting them extremely slow and also a massive target that's easy to hit and weak to EMP. Additionally if your target is just parading around a decked out combat mech as a non-antag for no reason, you should probably bring that up to admins.

Yeah okay that tells me you've never dealt with mechs before tell me exactly what a emp is going to do when they're loaded up with supercharged yellows, unless you have a ion carbine you wont do anything to them which come in limited amounts or from sec lathe + firing pin, oh also a entire clip of a ion carbine will not kill a Durand, giving them time to repair.

 
> That aside, if you care more about making your target dead than interacting with them and primarily rely on reagents and stuns to do so, you sure do sound like the fun kind of antagonist player everyone enjoys having around.

Yes because its so easy to interact with a guy sitting in a mech. good on you for making assumptions, makes me wonder how much you even play yourself.

**Comment by Rukofamicom** [comment]
>  i cant tell if you're trying to be funny with the @ or if you just can't read my name.

The latter lmao
You've commented on so few PRs I wasn't sure you'd even see my response if I didn't ping.

> The voodoo doll for example was a great tool that could also be used for RP gimmicks which you decided that miners should no longer have

The voodoo doll is a really great item, but I couldn't see any legitimately good use for it outside of antagonistic play due to its sheer power. That's why I kept it in the game, but shifted it to traitor uplinks - though you're right it's not available for shaft miners specifically. 

> Its almost like i suggested several other changes instead of just removing it I dont have a problem with it being changed for the better but outright removing it is like I said, just removing fun.

To give a little additional backstory - this PR was made in response to a discord conversation. One that started with *someone else* saying "I'm going to make a PR that completely removes the die of fate" because they are distraught by the ability to infinitely roll for chances at wizard. Everything within this PR is a compromise on that conversation to prevent a total removal by someone with the power and influence to potentially do so. I don't just come out of the woodwork and decide to remove fun - all of my nerfs are a result of conversations with others, and the vast majority of them are trying to placate someone who has a much more ham-fisted approach. *That information rarely makes it into the PR though*. 

That same conversation is why we have a total of three PRs addressing the same issue and fear of having the item yoinked entirely.
* This PR makes a 20 possible to roll but changes its effect to an immediate re-roll + another free roll after. 
* #8665 keeps wizard but makes it impossible to refresh the die, completely removing a different and cool interaction. I do not support this PR and would be pretty unhappy if it merged. 
* #8671 keeps wizard and refresh, but removes the 20 roll after refreshing. I support this PR and would be fine with it as an alternative - I still like my approach to having the greed ruin give a slightly rigged die more though. 

> i dont wanna derail the PR to talk about mechs

Yeah I'll leave that one alone for here - you can ping me on discord if you want to talk through this one, but I generally don't touch departments I don't play myself. I don't like messing with things I don't feel I fully understand after countless experiences with others suggesting ham-fisted changes on departments they don't know. This is why despite all my hate for botany I have never directly touched it myself.

**Comment by itsmeow** [comment]
I am most happy with this one out of the three PRs open. Players should not have access to the ability to roll wizard from lavaland loot but the reroll is an interesting possibility and removing it sucks.

**Comment by PowerfulBacon** [comment]
Closing in favour of https://github.com/BeeStation/BeeStation-Hornet/pull/8665
Reading through this PR is half convincing me that the dice shouldn't be a thing at all, even if you can't roll wizard being able to infinitely roll the dice is still a problem.
